### PR TITLE
 Aishwaryak - Equipment_List_View_Frontend

### PR DESCRIPTION
**Pull request by Aishwaryak01**
Description
This PR is about the EquipmentsList View from Phase 2 with Sorting based on project name and item name, Filtering based on selected values of the project and equipment, and Modal implementation for viewing update and purchase history.
Implements # (WBS) 9.1.1

Related PRS (if any):
This frontend PR is related to the #740 backend PR. (https://github.com/OneCommunityGlobal/HGNRest/pull/740)

Main changes explained:
Created components for Equipments

How to test:
1. check into the current branch
2. do npm install and npm run start: local to run this PR locally
3. Clear site data/cache
4. log as an admin user
5. verify all the table functionalities like open/close modals, sorting, and filtering for equipments view

image

image

image

image

image

### CODE REVIEW

**Review by 666saofeng — CHANGES_REQUESTED** [review]
Hi @Aishwaryak01, It seems that the equipment's page cannot be rendered on my side. Please refer to the video for the details

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62363348/08e5ff33-082a-4ca3-b6f9-b6a34b2d968a

**Review by peterson337 — DISMISSED** [review]
HI Aishwaryak01, I tested your application, and everything worked as expected. Good job!

one comunnity

one comunnity

**Comment by mianmiantea2019** [comment]
Hi @Aishwaryak01. The equipment page cannot be rendered on my side, the 'equimentslist' is undefined.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/59427068/6d39895f-a252-4384-8fb4-4e4928d737a5

**Review by ShengjieMao — COMMENTED** [review]
Hi, I also met errors when loading the equipments page. The page doesn't rendered properly, and received error messages in console saying missing properties. Most of these properties are suggested to be in the equipmentslist file. The error message looks like:
PR_1937

**Review by meetpadhiar4 — CHANGES_REQUESTED** [review]
Hi, I reviewed your PR and it is giving errors while rendering the equipments page. Please refer the video attached for your reference.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/b59fe352-3cd5-474c-8652-e5b7c6fd8dc7

**Comment by Aishwaryak01** [comment]
@mianmiantea2019 @ShengjieMao @666saofeng @meetpadhiar4  Can you check now? I have made the necessary changes.

File: src/actions/bmdashboard/equipmentActions.js [hunk 1]
```diff
@@ -10,4 +12,31 @@ export const addEquipmentType = async (body) => {
...
+    payload
+  }
+}
+
+export const fetchAllEquipments = () => {
+  return async dispatch => {
+    axios.get(ENDPOINTS.BM_EQUIPMENTS)
+      .then(res => {
+        dispatch(setEquipments(res.data))
+        console.log("Equipment is", res);
```
Comment (swathyjayaseelan) [inline src/actions/bmdashboard/equipmentActions.js:36]: Please comment out or remove the log statement.

**Review by swathyjayaseelan — DISMISSED** [review]
The changes look good.
 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13806470/c35801b0-a15b-4765-90e1-e7d67521f34a

**Comment by TareqMia** [comment]
Hi @Aishwaryak01, I tested out this PR and the equipment table looks good and everything functions as it is supposed to (sorting, modals). However, I did notice that when I try to view the purchase modal for:
Screenshot 2024-02-17 at 9 46 10 PM 

the app crashed. I think it might have something to do with the actual dummy data being faulty. Just wanted to bring that to your attention. Other than that, everything else looks good 👍 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/39747003/b7a7e154-5216-42da-b0db-3d1b03bc6be6

**Comment by Aishwaryak01** [comment]
@TareqMia : I cross-checked the PR again and as you said found out that there was some issue in the collection. So no changes are required in the code

**Review by XiaohanMeng — COMMENTED** [review]
Hi, I tested this PR with an admin account and encountered the same issue TareqMia mentioned. I verified all functionalities for the equipment view, including modals, sorting, and filtering. However, when clicking the purchase button on the second row, the same error occurred. Please see the attached video.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113215317/4df6cafc-2f2c-496d-beb3-b330c795e062

**Comment by Aishwaryak01** [comment]
Can you check now as I have fixed the issue in the collection?

**Review by skaush21 — DISMISSED** [review]
- logged in to admin account, navigated to bmdashboard
- logged in again, navigated to equipments
- tested project/name tabs, clicked on each view button, tested the drop down menus
- all features work
- the issue mentioned in the previous comment has been fixed


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/141872402/610ffcc3-8fd3-473f-8b0c-3bfe720a38a0

**Review by brandta-1 — DISMISSED** [review]
# Review
Tested all sorting arrows, dropdown menus, and view buttons. All components were functioning correctly.

# Video


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/116298512/d8d11385-8586-4d6a-a9b3-4dd38cbfea41

**Review by meetpadhiar4 — DISMISSED** [review]
Hi, I reviewed the PR again and the code now works correctly. Please refer the video attached.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/ed8d135d-3276-4f3b-b0bc-c4d79a19cc36

**Review by XiaohanMeng — DISMISSED** [review]
Hi @Aishwaryak01, I retested this PR, and all the previously mentioned issues have been resolved. Great job!
However, I've encountered some minor issues related to sorting:
* Initially, when I enter the page, the arrow next to 'Project' points upwards, indicating that the projects are sorted in ascending order. However, after clicking the 'Name' column's arrow a few times and then returning to click the 'Project' column's arrow, the 'Project' arrow points downwards while still sorting the content in ascending order. A similar issue occurs with the 'Name' column.
* The filters for 'project' and 'name' work well initially, displaying data correctly. But, when I click the sorting arrows for 'Name' or 'Project', all data appears, ignoring the applied filters. I'm not sure if this is intentional.
Please refer to the attached video.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113215317/92a3c44e-6931-418a-b1f8-9e6db24229ee

**Comment by Weiyao-Li** [comment]
Hi, while reviewing your pr, I encountered sever side error. Please have a check. I have attached relevant screenshots about the situation.

image
image
image

**Comment by DiegoSalas27** [comment]
Great work. Everything works as expected and your PR is clean and contains only the changes regarding your functionality. I attached a video showcasing how everything works:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/26840798/1c7361f4-26b0-4dfb-91cd-0b568daa8e69

**Review by DiegoSalas27 — DISMISSED** [review]

**Review by Weiyao-Li — DISMISSED** [review]
Hi, while reviewing your pr, I encountered sever side error. Please have a check. I have attached relevant screenshots about the situation.

**Review by SRamen1999 — COMMENTED** [review]
While reviewing this PR, I encountered sever side error. Also when I am in the http://localhost:3000/bmdashboard/equipments route no data is being loaded because I
 am getting a 404 error.

Screen Shot 2024-03-07 at 9 06 28 PM

Screen Shot 2024-03-07 at 9 06 57 PM

**Review by Gugor99 — COMMENTED** [review]
### Report
Hi,
I tried to review your PR but compilation failed and I got an error message in the console. Please see screenshot for reference.
### Screenshot
Cattura1

**Comment by rajnada** [comment]
Hi Aishwariya,

Great Job! All the functionality that you mentioned is working fine.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/25640436/efff3b18-5aff-4921-bdff-42d8346de800

**Review by Coops023 — DISMISSED** [review]
Hi, i have reviewed the functionality of this PR and all seems to work well. 

Screenshot 2024-03-08 at 20 51 14

**Comment by Aishwaryak01** [comment]
@Gugor99 @Weiyao-Li @SRamen1999 : I re-tested and the page is rendered successfully on my side. I think it is an intermittent issue as others can view it without any issue.

**Comment by SanketKaware** [comment]
Followed all the mentioned steps and the changes look good to me.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/17793708/185284be-b752-4446-a222-5630d57f0972

**Review by Sarthak32 — COMMENTED** [review]
I am not able to see  the projects in the  equipments page:
Screenshot 2024-03-09 133030
Screenshot 2024-03-09 133103

**Review by bhuvan-dv — DISMISSED** [review]
Followed all the mentioned steps and the changes look good to me.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67942070/8bd8715d-4a76-4b0a-ad23-9eb84d5dc5ab

**Review by YHX6 — DISMISSED** [review]
It works correctly.
Screenshot 2024-03-09 at 3 43 44 PM

**Review by psharma1984 — APPROVED** [review]
Hi @Aishwaryak01 Everything works as expected. Approving !!!
a
c
d
Screenshot 2024-03-10 012904

**Review by sameersdeshpande — DISMISSED** [review]
Hello Aishwarya!
I tested the dashboard on my end. It is working fine as expected.
But I want to check that when I sort project and equipment, and then click on ascending or descending. It lists all the items. Should not it be listing only the itmes of the sorted list.
May I know if this is the thing to be considered? Uploading the video here.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87390339/52da09ea-f178-4875-8263-6630ec9a2c83


Thanks

**Review by 20chen-7 — COMMENTED** [review]
Every function looks great. just one thing, npm start for the frontend will cause test error
Screenshot 2024-03-17 at 12 09 36 AM
Screenshot 2024-03-17 at 12 07 23 AM
Screenshot 2024-03-17 at 12 07 55 AM
Screenshot 2024-03-17 at 12 08 04 AM
Screenshot 2024-03-17 at 12 08 18 AM

Screenshot 2024-03-17 at 12 09 03 AM
3-0353-4e9d-8593-5d6894907596">

**Review by iven-yao — DISMISSED** [review]
- [x] open edit/update/purchase modal correctly
- [x] close edit/update/purchase modal correctly
- [x] sorting on project/ name works correctly
- [x] filtering on project/ equipment works correctly  

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/25358966/34927ad5-5bc0-4d7f-8a85-371021e4f5d0

**Review by hetvi11110 — DISMISSED** [review]
I have looked over and tested this PR as it was mentioned. It is working perfectly on my local server. Here is a reference video.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13557067/70e404c2-0741-4589-a7ed-eed38894f588

**Review by ImzIssa — DISMISSED** [review]
Hey, I've tested your PR and all your changes work as expected. 

Screenshot (86)
Screenshot (85)

**Review by kylepham — DISMISSED** [review]
Functionalities work as expected. Here is the testing video.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/35129229/150333a8-894b-4374-b7ad-51f64d7e11dc

However, I noticed that the first click on either "Project" or "Name" doesn't yield any changes. I think it was because the data was already sorted that way. 2 things that I want to point out:
- If the initial data is not sorted, I think there shouldn't be any sorting direction indicators next to column names. They should be there once the user triggers.
- Plus, there should be only one column being sorted at a time. I've found [an example](https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/), you can check it out.

These improvements should improve UX, imo. Overall, great work 👍🏻

**Comment by Gayathr-2012** [comment]
Hi, 
I have tested your PR. I followed the steps you mentioned, and it works as expected.  Good job.
image


image

image

image

**Review by Jaiwin-4998 — DISMISSED** [review]
Hi I just reviewed you PR and all the features are working as intended. I am attaching the video file for the reference. Please let me know if I missed something to check.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62836815/082aea22-d912-4223-af9e-85d48f1cd415

**Review by Parth-tech — DISMISSED** [review]
Hey Aishwarya,

Everything works correctly. Great job done. I have just one feedback. 

`Feedback:`
- When you click on the column name to sort i.e `Name` or `Project` column, and if you do not move your cursor the tooltip still displays the same message. This is very confusing. Better if you solve this. 
- Meaning if the above statement is ambiguous: You hover on `Name` Column. The tooltip displays `sort name asc`. Once you click on it then the `Name` column will be sorted in ascending order. Now if you do not move your cursor the tooltip still continues to display `sort name asc`. Hence, after the click we can simply get rid of the tool tip and display when we re-hover over it


Screenshot 2024-04-06 at 7 35 10 PM
Screenshot 2024-04-06 at 7 34 35 PM
Screenshot 2024-04-06 at 7 34 45 PM

Screenshot 2024-04-06 at 7 34 55 PM

**Comment by MQChen211** [comment]
Hi, I tested your PR and it works as expected.
截屏2024-04-09 14 03 31
截屏2024-04-09 14 48 09
截屏2024-04-09 14 48 17
截屏2024-04-09 14 04 05
3d">

**Review by j-corp-25 — COMMENTED** [review]
Hi @Aishwaryak01 ,

I tested as instructed but im not seeing a list of items. I checked the endpoint and its returning ok. But I still dont see the items as listed on your pictures

Please see below:

TestingEquipment

**Review by cvtqx — DISMISSED** [review]
I've tested this PR from an admin account and all features work as described. There is a little delay happening when a tooltip is displayed on hover and it is not disappearing once the arrow is clicked (someone has mentioned this already). this should probably be addressed in a different PR as it seems just a tiny cosmetic defect. Overall great job!
See video of testing:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/107726913/9518c286-21c4-44d7-bfc5-ed5b181895cd

**Comment by kaushikMreddy293** [comment]
Hi, 
I tried to test this PR but, the data is not showing up in the table for me, can anyone share the URL or update if I've missed something in the implementation

image

**Comment by KaushikMreddy** [comment]
Hi, 
I tried to test this PR but, the data is not showing up in the table for me, can anyone share the URL or update if I've missed something in the implementation

image

**Comment by one-community** [comment]
Thank you all, merging!

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**